aclk/agent_cloud_link.c

```diff
@@ -129,6 +161,31 @@ struct aclk_query_queue {
     u_int64_t count;
 } aclk_queue = { .aclk_query_head = NULL, .aclk_query_tail = NULL, .count = 0 };
 
+/*
+ * After a connection failure -- delay in milliseconds
+ * When a connection is established, the delay function
+ * should be called with mode 0 to reset the fail count
+ */
+unsigned long int aclk_delay(int mode)
+{
+    static int fail = 0;
+
+    if (!mode) {
+        fail = 0;
+        return 0;
+    }
+
+    if (fail == 0) {
+        srandom(time(NULL));
+        fail++;
+        return 0;
+    }
+
+    unsigned long int delay = (unsigned long int)((powl(ACLK_DELAY_SEED, fail)) * 1000 + (random() % 1618));
```

Comment:
Is there any possibility of `fail` one day to become a number different of [0,1] where this interval only have integer numbers? The actual code says no, but it is possible to be changed in the future. Case the answer is `no`, please, remove completely the call to `powl`, because this function will only return either `ACLK_DELAY_SEED` or `1`, but these values can be received with a simple condition, instead to use a function that has a high cost to process(Probably it has a Taylor's series truncation or an algorithm that optimizes, but cannot be faster than an `if`).

Reply:
This is a bug -- there is a fail++ missing.. Will be fixed and be available in PR 8037

Reply:
All right, thank you for the explanation!  :)